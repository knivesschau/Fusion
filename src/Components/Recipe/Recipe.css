.RecipeMain {
    font-size: 1.10em;
}

.Recipe_Expand {
    text-align: center;
    border: 1px solid black;
    margin: 2.5em;
}

#ingredients-list {
    padding-left: 1em;
    margin: 1em;
}

#steps-list {
    padding-left: 1em;
    margin-right: .5em;
    margin: .5em;
}

#modify-recipe, #delete-recipe, #modify-recipe2, #delete-recipe2 {
   font-size: 1em;
   margin-left: auto;
   margin-right: auto;
   margin: .5em;
}
