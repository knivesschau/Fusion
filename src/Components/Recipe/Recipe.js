import React, { Component } from 'react';
import fusionContext from '../../fusionContext';
import { Link } from 'react-router-dom';
import config from '../../config';
import './Recipe.css';

export default class Recipe extends Component {    
    static defaultProps = {
        onDeleteFusion: () => {}
    };

    static contextType = fusionContext;

    handleDelete = e => {
        e.preventDefault();
        const fused_id = this.props.fused_id;

        fetch(`${config.API_ENDPOINT}/recipes/${fused_id}`, {
            method: 'DELETE',
            headers: {
                'content-type': 'application/json'
            },
        })
            .then(res => {
                if (!res.ok) {
                    return res.json().then(e => Promise.reject(e));
                }
                return null;
            })
            .then(() => {
                this.context.deleteFusion(fused_id);
                this.props.onDeleteFusion(fused_id);
            })
            .catch(error => {
                console.error({error})
            });
    };

    renderCuisines() {
        const {fuse_cuisine, base_cuisine} = this.props;

        if (fuse_cuisine === null || fuse_cuisine === 'None') {
            return (
                <h3 id="cuisine-styles">{base_cuisine}</h3>
            );
        }
        else {
            return (
                <h3 id="cuisine-styles">{base_cuisine}, {fuse_cuisine}</h3>
            );
        }
    };

    render() {

        const {fused_id, fused_name, date_modified, fuse_ingredients, fuse_steps} = this.props;
        
        // convert data into presentable format on client
        const convertIngredients = Object.values({fuse_ingredients}).join();
        const fused_ingredients = convertIngredients.split("\n");
        
        // convert data into presentable format on client
        const convertSteps = Object.values({fuse_steps}).join();
        const fused_steps = convertSteps.split("\n");

        return (
            <section className="RecipeMain">
                
                <div className="Recipe_Expand">

                    <h2 id="recipe-title">{fused_name}</h2>
                    
                    {this.renderCuisines()}
                    
                    <h4 id="date-changed">Last Modified: {new Date(date_modified).toLocaleDateString()} </h4>

                    <Link to={`/edit-recipe/${fused_id}`}>
                        <button id="modify-recipe" type="button">Modify</button>
                    </Link>

                    <button id="delete-recipe" type="button" onClick={this.handleDelete}>Delete</button>

                    <div className="Ingredients_Section">
                        <h4 id="ingredients-title"><u>Ingredients</u></h4>
                        
                        <ul>
                            {fused_ingredients.map((fuse_ingredients, i) => {
                                return (
                                    <li id="ingredients-list" key={i}>{fuse_ingredients}</li>
                                );
                            })}
                        </ul>
                    </div>
                
                    <div className="Steps_Section">
                        <h4 id="steps-title"><u>Steps</u></h4>
                        
                        <ol>
                            {fused_steps.map((step, i) => {
                                return (
                                    <li id="steps-list" key={i}>{step}</li>
                                );
                            })}
                        </ol>
                    </div>

                </div>

            </section>
        );
    };
};