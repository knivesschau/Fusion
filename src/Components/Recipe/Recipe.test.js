import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Recipe from './Recipe';

describe ('Recipe Component', () => {
    it ('Renders the complete Recipe Component and all nested Components', () => {
        const wrapper = shallow(<Recipe/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});