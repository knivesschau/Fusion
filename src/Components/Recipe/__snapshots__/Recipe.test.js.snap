// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Component Renders the complete Recipe Component and all nested Components 1`] = `
<section
  className="RecipeMain"
>
  <div
    className="Recipe_Expand"
  >
    <h2
      id="recipe-title"
    />
    <h3
      id="cuisine-styles-both"
    >
      , 
    </h3>
    <h4
      id="date-changed"
    >
      Last Modified: 
      Invalid Date
       
    </h4>
    <Link
      to="/modify-recipe/undefined"
    >
      <button
        id="modify-recipe"
        type="button"
      >
        Modify
      </button>
    </Link>
    <button
      id="delete-recipe"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
    <div
      className="Ingredients_Section"
    >
      <h4
        id="ingredients-title"
      >
        <u>
          Ingredients
        </u>
      </h4>
      <ul
        id="display-ingredients"
      >
        <li
          id="ingredients-list"
          key="0"
        />
      </ul>
    </div>
    <div
      className="Steps_Section"
    >
      <h4
        id="steps-title"
      >
        <u>
          Steps
        </u>
      </h4>
      <ol
        id="display-steps"
      >
        <li
          id="steps-list"
          key="0"
        />
      </ol>
    </div>
  </div>
</section>
`;
